docs/apache-airflow/authoring-and-scheduling/deferring.rst

```diff
@@ -253,6 +253,73 @@ These parameters can be mapped using the ``expand`` and ``partial`` methods. Not
         trigger_kwargs=[{"moment": timedelta(hours=2)}, {"moment": timedelta(hours=2)}]
     )
 
+Exiting deferred task from Triggers
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+ .. versionadded:: 2.10.0
+
+If you want to exit your task directly from the triggerer without going into the worker, you can specific operator level attribute ``end_from_trigger`` with the attributes to your deferrable operator other discussed above.
+
+Triggers can have two options: they can either send execution back to the worker or end the task instance directly. If the trigger ends the task instance itself, the ``method_name`` does not matter and can be ``None``. Otherwise, provide the name of the ``method_name`` that should be used when resuming execution in the task.
+
+.. code-block:: python
+
+    class DateTimeSensorAsync(DateTimeSensor):
+        def __init__(self, end_from_trigger: bool = True, **kwargs) -> None:
+            super().__init__(**kwargs)
+            self.end_from_trigger = end_from_trigger
+
+        def execute(self, context: Context) -> NoReturn:
+            self.defer(
+                method_name="execute_complete",
+                trigger=DateTimeTrigger(
+                    moment=timezone.parse(self.target_time), end_from_trigger=self.end_from_trigger
+                ),
+            )
+
+        def execute_complete(self, context, event=None) -> None:
+            return None
```

Comment:
```suggestion
    class DateTimeSensorAsync(DateTimeSensor):
        def __init__(self, **kwargs) -> None:
            super().__init__(**kwargs)
            self.end_from_trigger = True

        def execute(self, context: Context) -> NoReturn:
            self.defer(
                method_name=None,
                trigger=DateTimeTrigger(
                    moment=timezone.parse(self.target_time), end_from_trigger=self.end_from_trigger
                ),
            )
```

Let’s make this example simple for people. If users want their sensor to support both modes, I think they should be smart enough to add the argument.

(We should also add text to tell users this sensor _always_ exit from trigger.)

Reply:
I have changed the example completely now with `WaitFiveHourTrigger` and `WaitFiveHourSensorAsync`.